components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1067,23 +1083,50 @@ private String doUserAuthz(HttpServletRequest request, String sessionDataKey,
 
         } else if ((OAuthConstants.Prompt.NONE).equals(oauth2Params.getPrompt())) {
             //Returning error if the user has not previous session
-            if (sessionDataCacheEntry.getLoggedInUser() == null) {
+            if (user == null) {
+                errorResponse = EndpointUtil.getErrorRedirectURL(
+                        OAuthProblemException.error(OAuth2ErrorCodes.LOGIN_REQUIRED), oauth2Params);
                 return errorResponse;
+            }
+            String idTokenHint = oauth2Params.getIDTokenHint();
+            //Evaluate the id_token_hint value if it is associate with the request.
+            if (StringUtils.isNotEmpty(idTokenHint)) {
```

Comment:
Shouldn't we at least validate JWT ? signature validation ?

Reply:
introduced  validateIdToken() to check this